src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -738,7 +749,12 @@ namespace xclcpuemhal2 {
       if ( mLogStream.is_open() ) {
         verbose = true;
       }
-
+      setDeviceProcessStarted(true);
+      //Thread to fetch messages from Device to display on host
+      if(mMessengerThreadStarted == false) {
+        mMessengerThread = std::thread(xclcpuemhal2::messagesThread,this);
+        mMessengerThreadStarted = true;
+      }
```

Comment:
Avoid obfuscated code and cryptic API. Why not just uing the thread status instead of adding a `mMessengerThreadStarted`?
```suggestion
      if (!mMessengerThread.joinable())
        mMessengerThread = std::thread([this] { messagesThread(); });
```

Reply:
Updated according. Used Lambda for creating the messagesThread.